packages/bridge-ui-v2/src/components/Bridge/NFTBridgeSteps/ImportStep.svelte

```diff
@@ -203,7 +204,15 @@
       canProceed = false;
     }
   } else if (importMethod === 'scan' && $selectedNFTs && $selectedNFTs.length > 0 && $destinationChain && scanned) {
-    canProceed = true;
+    if ($selectedNFTs && $selectedNFTs[0].type === TokenType.ERC1155) {
```

Comment:
```
 function validateManualImport() {
    if (detectedTokenType === TokenType.ERC721 || detectedTokenType === TokenType.ERC1155) {
      return (
        addressInputState === AddressInputState.VALID &&
        idInputState === IDInputState.VALID &&
        (detectedTokenType === TokenType.ERC1155 ? $enteredAmount > BigInt(0) : true) &&
        isOwnerOfAllToken
      );
    }
    return false;
  }

  function validateScanImport() {
    if ($selectedNFTs && $selectedNFTs.length > 0 && $destinationChain && scanned) {
      if ($selectedNFTs[0].type === TokenType.ERC1155) {
        return $enteredAmount > BigInt(0);
      } else if ($selectedNFTs[0].type === TokenType.ERC721) {
        return true;
      }
    }
    return false;
  }
  $: canProceed =
  importMethod === 'manual' && validateManualImport()
    ? true
    : importMethod === 'scan' && validateScanImport()
    ? true
    : false;
```
    
    I think you should do something similar to this to make the code look clearer. This one looks too bad

Reply:
Absolutely! 👍  There are still a couple of places where functions grew too much and should be deconstructed into smaller individual parts. Good catch, will improve this